### DISCUSSION TITLE
Nuxt Global Middleware

### DISCUSSION THREAD

**Discussion by WailanTirajoh** [nuxt/nuxt#25061] [Questions]
I've create nuxt modules for internal project, inside that i created global middleware, named `auth`. It is added via `addRouteMiddleware` function through `plugin.ts` as global

```js
addRouteMiddleware(
    "auth",
    async (to) => {
      // some code
    },
    { global: true }
  );
```
  
In other nuxt app that uses this nuxt modules is also injected this auth automaticaly (as expected), but i want to disable this `auth middleware` on specific page. Is there any chance i can do this? Thanks 🐎

**Comment by xjccc** [comment]
use `useRuntimeConfig`?

```
// nuxt.config.ts
runtimeConfig :{
   public: {
      disabledRouteNames: ['index', 'sub-page']
   }
}
// plugins

const config = useRuntimeConfig()
addRouteMiddleware(
    "auth",
    async (to) => {
      if (config.public.disabledRouteNamse.includes(to.name)) return
    },
    { global: true }
  )

```

**Reply by WailanTirajoh** [reply]
Thanks, so we're need to update the global middleware to add guard clause & it will still registered as active middleware.

**Comment by rezansrv** [comment]
In Nuxt.js, when you use `addRouteMiddleware` with the `global: true` option, it means that the middleware will be applied to all routes globally. However, you can exclude certain routes from this global middleware by using the `except` option in the `nuxt.config.js` file.

Here is an example of how you can exclude a specific route from the global middleware:

```javascript
// nuxt.config.js

export default {
  // Other config options...

  router: {
    middleware: ['auth'],
  },

  globalMiddleware: [
    async (to, from, next) => {
      // Global middleware logic
      console.log('Global middleware');
      next();
    },
  ],

  auth: {
    except: ['/exclude-this-route'],
  },
};
```

In this example:

- `router.middleware` is used to add the global middleware (`auth`) to all routes.
- `globalMiddleware` is used for defining global middleware logic. This middleware will be applied to all routes globally.
- `auth` module configuration includes the `except` option to exclude the middleware for the specified route (`/exclude-this-route` in this case).

By using the `except` option in the `auth` module configuration, you can specify the routes where the middleware should not be applied. Adjust the route path as needed for your specific case.

**Reply by septatrix** [reply]
How would one do that with `defineNuxtRouteMiddleware` under `middleware/`? That does not seem to have any name?
